src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java

```diff
@@ -47,16 +47,21 @@ public HttpSourceBridgeEndpoint(Vertx vertx, HttpBridgeConfig bridgeConfigProper
     public void handle(Endpoint<?> endpoint) {
         HttpServerRequest httpServerRequest = (HttpServerRequest) endpoint.get();
 
-        messageConverter = new HttpJsonMessageConverter();
 
         // split path to extract params
         String[] params = httpServerRequest.path().split("/");
 
         // path is like this : /topics/{topic_name}, topic will be at the last position of params
-        String topic = params[params.length - 1];
+        String topic = params[2];
+
+        if (params.length == 5) {
```

Comment:
Why this distinction? what is param number 4? If it's about partition-id on the URL we should remove that, it's not part of this PR.